src/Routes/DynamicRoute/DynamicRoute.tsx

```diff
@@ -20,12 +20,15 @@ const DynamicRoute: React.FC<DynamicRouteProps> = ({ location }) => {
   const dynamicRoutePages = useExtensions<DynamicRoutePage>(isDynamicRoutePage);
   React.useEffect(() => {
     if (location) {
-      const [, , app] = location.pathname?.split('/') || [];
-      if (app) {
+      const [, first, second] = location.pathname?.split('/') || [];
```

Comment:
Yup, the first part is `hac`, then we have a rest of the path. I was blidnly expecting first partial for flat nav structure and second for navigation in sections (second level nav). The problem with comparing the `rest` is what about details? What if we have url `hac/item` for the list and `hac/item/123456789` for the detail. What if the detail will be managed by the extension itself and user would refresh it, they'd be at best be navigated to the parent route (`hac/item`) instead of the detail one.

This is probably a lack of my knowledge of all possibilities in the extensions, but this seemed most logical and direct approach. I have no objections to consuming the entire URL and checking for possible routes though.

Reply:
DevConsole Package in OpenShift Console still uses static plugins to setup their routes. However, once converted over to the dynamic extension model, [this will be a use-case](https://github.com/openshift/console/blob/master/frontend/packages/dev-console/src/plugin.tsx#L213) of issue in this "only look at the first two parts of the route":
```
path: ['/k8s/ns/:ns/buildconfigs/~new/form', '/k8s/ns/:ns/buildconfigs/:name/form'],
```

This particular use-case for `buildconfigs` may not translate fully into HAC, but this design will be something we need to concern ourselves with. The extension model allows them to define however they want things to go. They are not limited to a two route depth system.

> What if we have url hac/item for the list and hac/item/123456789 for the detail.
As I covered in my [comment below](https://github.com/RedHatInsights/hac-core-frontend/pull/16#discussion_r743886128), paths are not (always) going to be simple, as there is no expectation for them to be. Nothing stops someone from doing `hac/item` quite simply, but it has nothing to do with navigation structure. This could be navigated from a collapsible nav item.